### PR TITLE
[Integration]: Google Fit Integration

### PR DESCRIPTION
**Pull request by sid2002CN**
## Description

Implemented the Google Fit API Integration with MindsDB handler. Got the basics done: implement the select query to get step data from google fit with a start date and end date.

**Fixes** #3818

## Type of change

(Please delete options that are not relevant)

- [x] ⚡ New feature (non-breaking change which adds functionality)
- [x] 📄 This change requires a documentation update

### What is the solution?
Get the steps data from google fit API, which can be queried with the date range specified.

### Screenshot:
Connection:
image

Select the query:
fa4c92ac895edf113a4f65e986d59b1

### TODO:
Add the functionality to deal with calorie data and heartbeat data from google fit.
Add more documentation.
Add more supported queries.
We will try to finish these in the next few weeks.

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation, or created issues to update them.
- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] I have shared a short loom video or screenshots demonstrating any new functionality.

### CODE REVIEW

**Comment by ZoranPandovski** [comment]
recheckcla

**Comment by mindsdbadmin** [comment]
**CLA Assistant Lite bot:** <br/>Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you all sign our [Contributor License Agreement](https://github.com/mindsdb/mindsdb_native/blob/stable/assets/contributions-agreement/individual-contributor.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.<br/>
   - - -
   I have read the CLA Document and I hereby sign the CLA
   - - -
   **2** out of **3** committers have signed the CLA.<br/>:white_check_mark: @yurihwang<br/>:white_check_mark: @2170chm<br/>:x: @sid2002CN<br/><sub>You can retrigger this bot by commenting **recheck** in this Pull Request</sub>

File: mindsdb/integrations/handlers/google_fit_handler/google_fit_tables.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        filters = []
+        steps = {}
+        now = int(round(time.time() * 1000))
+        one_year = 31536000000
+        one_month = 2629746000
+        for op, arg1, arg2 in conditions:
+            if op == 'or':
+                raise NotImplementedError(f'OR is not supported')
+            if arg1 == 'date':
+                print(f'args : {arg2} 2222222222222222222')
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/google_fit_handler/google_fit_tables.py:32]: Can we remove prints please?

File: mindsdb/integrations/handlers/google_fit_handler/google_fit_handler.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        self.api = None
+        self.is_connected = False
+
+        aggregated_data = GoogleFitTable(self)
+        self._register_table('aggregated_data', aggregated_data)
+
+    def connect(self) -> Resource:
+        if self.is_connected is True and self.api:
+            return self.api
+        if len(self.connection_args) == 7:
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/google_fit_handler/google_fit_handler.py:48]: Do we create a new file from the user's provided parameters? If so, maybe we can add an option for the user just to send one parameter, which will be the credentials.json file

File: mindsdb/integrations/handlers/google_fit_handler/google_fit_handler.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        Args:
+            query (ASTNode): sql query represented as AST. May be any kind
+                of query: SELECT, INSERT, DELETE, etc
+        Returns:
+            DataFrame
+        """
+        self.connect()
+        if method_name == 'get_steps':
+            val = self.get_steps(params['start_time'], params['end_time'])
+            print(val)
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/google_fit_handler/google_fit_handler.py:144]: Please remove the print

File: mindsdb/integrations/handlers/google_fit_handler/google_fit_tables.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import pytz
+import time
+from tzlocal import get_localzone
+
+class GoogleFitTable(APITable):
+
+    def time_parser(self, args) -> int:
+        ymd = args.split('-')
+        epoch0 = datetime.datetime(1970, 1, 1, tzinfo=pytz.utc)
+        time = pytz.timezone(str(get_localzone())).localize(datetime.datetime(int(ymd[0].rstrip()), int(ymd[1].rstrip()), int(ymd[2].rstrip())))
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/google_fit_handler/google_fit_tables.py:15]: Can you add docstring to this method so users can know what this method does? Also can we simplify this?

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]
Thanks, please check the above comments

File: mindsdb/integrations/handlers/google_fit_handler/google_fit_handler.py (hunk 2 above)
Comment (2170chm) [reply mindsdb/integrations/handlers/google_fit_handler/google_fit_handler.py:48]: Hey! So if we add an option to just accept a credentials.json file, where would we want the user to put it? Do you think it is better to put it under mindsdb's topmost directory or within the Google Fit handler's directory, same as where the file would be generated now?

**Comment by sid2002CN** [comment]
I have read the CLA Document and I hereby sign the CLA

File: mindsdb/integrations/handlers/google_fit_handler/google_fit_tables.py (hunk 1 above)
Comment (sid2002CN) [reply mindsdb/integrations/handlers/google_fit_handler/google_fit_tables.py]: Sorry about that we pushed some of our debugging print statements accidentally, and I just delete all of them.

File: mindsdb/integrations/handlers/google_fit_handler/google_fit_handler.py (hunk 2 above)
Comment (ZoranPandovski) [reply mindsdb/integrations/handlers/google_fit_handler/google_fit_handler.py:48]: @2170chm You can check the Google Big query example

*    [defining](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/handlers/bigquery_handler/bigquery_handler.py#L156)
*    [using](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/handlers/bigquery_handler/bigquery_handler.py#L38)

File: mindsdb/integrations/handlers/google_fit_handler/google_fit_tables.py (hunk 4 above)
Comment (sid2002CN) [reply mindsdb/integrations/handlers/google_fit_handler/google_fit_tables.py:22]: Just done the comment.

File: mindsdb/integrations/handlers/google_fit_handler/google_fit_handler.py (hunk 2 above)
Comment (sid2002CN) [reply mindsdb/integrations/handlers/google_fit_handler/google_fit_handler.py]: @ZoranPandovski Scott and I have add the option to use credentials.json file. We have also fixed a few bugs and tested on our end.

**Comment by ZoranPandovski** [comment]
Thanks team

**Review by ZoranPandovski — APPROVED** [review]

**Pull request merged**